champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/sword/Ride.java

```diff
@@ -95,6 +103,8 @@ public void run() {
                 if (!horse.isDead()) {
                     horse.remove();
                     horseData.remove(player);
+                    activeHorses.remove(horse);
+
                 }
             }
         }.runTaskLater(champions, (long) (lifespan + (level - 1)) * 20L);
```

Comment:
Instead of having a bunch of `BukkitRunnable`s for each horse that is spawned through this ability, you should probably setup an `@UpdateEvent` (just search for how it's used in the rest of the project), loop through the alive horses and despawn those that are either dead or invalid.